client/state/data-layer/wpcom/gravatar-upload/index.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Internal dependencies
+ */
+import {
+	GRAVATAR_UPLOAD_RECEIVE,
+	GRAVATAR_UPLOAD_REQUEST,
+	GRAVATAR_UPLOAD_REQUEST_SUCCESS,
+	GRAVATAR_UPLOAD_REQUEST_FAILURE,
+} from 'state/action-types';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import {
+	bumpStat,
+	composeAnalytics,
+	recordTracksEvent,
+	withAnalytics,
+} from 'state/analytics/actions';
+
+export function uploadGravatar( { dispatch }, action ) {
+	const { email, file } = action;
+	dispatch( http( {
+		method: 'POST',
+		path: '/gravatar-upload',
+		body: {},
+		apiNamespace: 'wpcom/v2',
+		formData: [
+			[ 'account', email ],
+			[ 'filedata', file ],
+		],
+	}, action ) );
+}
+
+export function announceSuccess( { dispatch }, { file }, next, data ) {
+	// Return a Promise for testing purposes
```

Comment:
Yeah. A few hours away from this code made me realize that I was already turning the async `FileReader` into a sync process in the tests. I didn't feel good about the Promise, but yesterday I couldn't see a better way. Thanks for pushing back on that.